java/arcs/android/storage/service/BindingContextStats.kt

```diff
@@ -69,4 +75,37 @@ class BindingContextStatsImpl : BindingContextStatistics {
             }
         }
     }
+
+    override fun traceTransaction(tag: String?, block: () -> Unit) {
+        // TODO(ianchang): Inject Android system traces with [tag]
+        ++_transactions
+        block()
+        --_transactions
+    }
+
+    companion object {
+        private var _transactions = Transactions(Mutex())
+    }
+}
+
+data class Transactions(val mutex: Mutex? = null, var current: Int = 0, var peak: Int = 0) {
```

Comment:
Please provide KDoc for any new classes or public functions you create.

Reply:
np, will do.